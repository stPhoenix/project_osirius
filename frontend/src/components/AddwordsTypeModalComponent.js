import React from 'react';
import {Modal, ModalBody, Card, CardBody, CardTitle} from 'reactstrap';


export const AddwordsTypeModalComponent = (props) => {
    return (
        <Modal isOpen={props.modal} toggle={props.toggle_modal} className="p-0 align-self-center">
            <ModalBody className="p-0">
                <Card className="rounded-0">
                    <CardBody className="d-flex flex-column">
                        <CardTitle>Choose category</CardTitle>
                        
                            <form>
                                <select className="custom-form-control" id="learn_language" onChange={props.handleChange} name="category">
                                        {props.cats.map((category) => (<option key={category.name}>{category.name}</option>))}
                                </select>
                            </form>    
                        <button className="btn btn-link align-self-end" onClick={props.add_custom}>Add</button>
                    </CardBody>
                </Card>
            </ModalBody>
        </Modal>
    );
};