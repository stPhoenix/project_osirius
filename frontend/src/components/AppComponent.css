.theme-primary{
    background-color: #009be5;
}
.theme-light{
    background-color: #63ccff;
}
.theme-dark{
    background-color: #006db3;
}