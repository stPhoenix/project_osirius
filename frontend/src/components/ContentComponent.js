import React from 'react';


export const ContentComponent = (props) => {
    return(
        <div className="col-10 d-flex flex-column align-self-center align-items-center justify-content-center p-0">
            {props.children}
        </div>
    );
};