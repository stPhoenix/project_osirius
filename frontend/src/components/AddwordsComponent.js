import React from 'react';
import {Link, Switch, Route} from 'react-router-dom';
import typing from '../assets/typing.png';
import presets from '../assets/presets.png';
import './AppComponent.css';
import AddwordsType from '../containers/AddwordsType';
import AddwordsPresets from '../containers/AddwordsPresets';
import {ContentComponent} from '../components';
import PropTypes from 'prop-types';

const componentProps = {
  location: PropTypes.object.isRequired,  
};

export const AddwordsComponent = ({location}) => {
	if (location.pathname !== "/addwords"){
		return (<Switch>
					<Route path="/addwords/type" component={AddwordsType} />
					<Route path="/addwords/presets" component={AddwordsPresets} />
				</Switch>);	
	};
	return(
    <ContentComponent>
            <section className="d-flex flex-column align-items-center justify-content-center">
		      <div className="d-flex flex-column align-items-center justify-content-center">
                <img src={typing} alt="typing" style={{height:'7rem', width:'auto'}} />
		        <h4 className="opacity-50 text-center px-2 mb-4 mt-0">Type word. System will search word in database or in the internet.</h4>
		        <Link className="custom-btn theme-primary px-3 py-1 mb-5 black" to="/addwords/type" > Add words</Link>
              </div>
              <div className="d-flex flex-column align-top align-items-center justify-content-center">
                <img src={presets} alt="presets" className="" style={{height:'6rem', width:'auto'}} />
		        <h4 className="opacity-50 text-center px-2 mb-4 mt-0">Choose word from preset in database.</h4>
		        <Link className="custom-btn theme-primary px-3 py-1 mb-3 black" to="/addwords/presets" > Add words</Link>
              </div>
            </section>
	</ContentComponent>
		);
};

AddwordsComponent.propTypes = componentProps;