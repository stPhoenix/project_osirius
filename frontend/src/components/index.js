import {AppComponent} from './AppComponent';
import {CustomAlertComponent} from './CustomAlertComponent';
import {LoginComponent} from './LoginComponent';
import {NavBarComponent} from './NavBarComponent';
import {NavItemComponent} from './NavItemComponent';
import {NewsComponent} from './NewsComponent';
import {SignUpComponent} from './SignUpComponent';

export {
    AppComponent,
    CustomAlertComponent,
    LoginComponent,
    NavBarComponent,
    NavItemComponent,
    NewsComponent,
    SignUpComponent
};