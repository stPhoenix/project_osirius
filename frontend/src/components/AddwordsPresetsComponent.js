import React from 'react';
import './AppComponent.css';
import {WordsTableComponent, ContentComponent} from '../components';


export const AddwordsPresetsComponent = (props) => {
    return (<ContentComponent>
                <form className="col-12 col-lg-8 px-0">
                    <label htmlFor="categories" >Choose category:</label>
                    <select className="custom-form-control" id="categories" onChange={props.handleChange} name="category">
                                        {props.cats.map((category) => (<option key={category.name}>{category.name}</option>))}
                    </select>
                </form>
                <div className="d-flex align-self-end mt-3 px-0" >
					<button className="btn btn-link " onClick={props.add_selected}>Add selected</button>
					<button className="btn btn-link pr-0" onClick={props.add_all}>Add all</button>
				</div>
                <WordsTableComponent words={props.words} handleCheck={props.handleCheck} />
          </ContentComponent>);  
};