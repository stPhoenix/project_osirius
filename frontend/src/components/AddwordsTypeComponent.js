import React from 'react';
import './AppComponent.css';
import {CardDeck, Card, CardBody, CardText} from 'reactstrap';
import {AddwordsTypeModalComponent} from '../components';


export const AddwordsTypeComponent = (props) => {
	
	let data = null;
	if (props.global_word_search === true){
		data = (props.words.map(word => {
			return (<Card key={word.id+"words"} className="shadow-sm rounded-0">
                        <CardBody className="d-flex flex-column">
                            <CardText className="text-center">
                                <br />{word.name}
                                <br />{word.pronunciation}
                                <br />Category: [{word.category.map(category => (category.name+" "))}]
                                <br />{word.translation}
                            </CardText>
                            <button name={word.id} className="btn btn-link align-self-end" onClick={props.add_global}>Add</button>
                        </CardBody>
                    </Card>);
		}));
	} else if(props.google_translate_search === true){
		data += (props.words.map(word => {
			return (<Card key={word.id+"words"} className="shadow-sm rounded-0">
                        <CardBody className="d-flex flex-column">
                            <CardText className="text-center">
                                <br />{word.name}
                                <br />{word.pronunciation}
                                <br />{word.translation}
                            </CardText>
                        <button className="btn btn-link align-self-end" onClick={(e) => {props.choose_category(word.name, word.pronuciation, word.translation)}}>Add</button>
                        </CardBody>
                    </Card>);
		}));
	}
	
	return(<div className="col-10 d-flex flex-column px-0 justify-content-center align-items-center">
				<form className="d-flex flex-column w-100">
					<div className="form-group">
						<label htmlFor="search">Type word:</label>
						<input id="search" type="text" className="custom-form-control" value={props.search_word} name="search_word" onChange={props.handleChange}/>
					</div>
                    <button className="custom-btn theme-primary mb-2 align-self-center" onClick={props.search}>Search</button>
				</form>
                <CardDeck className="w-100">
                    {data}
                </CardDeck>
                <AddwordsTypeModalComponent add_custom={props.add_custom}
                                            handleChange={props.handleChange}
                                            cats={props.cats}
                                            modal={props.modal}
                                            toggle_modal={props.toggle_modal} />
		</div>);
};