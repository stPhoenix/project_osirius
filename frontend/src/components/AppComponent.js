import React from 'react';
import {Container, Col, Row} from 'reactstrap';
import './AppComponent.css';
import logo from '../assets/logo.png';
import NavBar from '../containers/NavBar';
import Login from '../containers/Login';
import Alerts from '../containers/Alerts';
import {Switch, Route} from 'react-router';
import News from '../containers/News';
import Logout from '../containers/Logout';
import SignUp from '../containers/SignUp';
import Settings from '../containers/Settings';
import {AddwordsComponent, MywordsComponent, WordstrainerComponent, AboutComponent, GetStartedComponent, PrivacyPolicyComponent, CookiePolicyComponent, DataCollectionComponent, FooterComponent} from '../components';
import Learnwords from '../containers/Learnwords';
import Policy from '../containers/Policy';


export const AppComponent = () => {
        return (
      <Container fluid className="bg-first">
        <Policy />
        <Row className="d-flex flex-column">
            <Col xs="12" className="px-0" style={{zIndex:1}}>
                <header className="theme-dark px-4 py-2">
                    <img style={{height:'2rem', width:'auto'}} src={logo} alt="logo" />
                </header>
            </Col>
            <Col xs="12" >
                <Row className="flex-column flex-lg-row">
                    <Col xs="12" lg="3" className="px-0" style={{zIndex:1}}>
                        <section className="theme-primary shadow-sm full-screen h-100">
                            <NavBar/>
                        </section>
                    </Col>
                    <Col xs="12" lg="9" className="px-0">
                        <main className="d-flex flex-column full-screen">
                            <Alerts />
                            <section className="d-flex justify-content-center align-items-center">
                            <Switch>
                                <Route exact path="/" component={News} />
                                <Route path="/news" component={News} />
                                <Route path="/login" component={Login} />
                                <Route path="/logout" component={Logout} />
                                <Route path="/signup" component={SignUp} />
                                <Route path="/settings" component={Settings} />
				                <Route path="/addwords" component={AddwordsComponent} />
                                <Route path="/mywords" component={MywordsComponent} />
                                <Route path="/learnwords" component={Learnwords} />
				                <Route path="/wordstrainer" component={WordstrainerComponent} />
                                <Route path="/about" component={AboutComponent} />
                                <Route path="/getstarted" component={GetStartedComponent} />
                                <Route path="/privacy_policy" component={PrivacyPolicyComponent} />
                                <Route path="/cookie_policy" component={CookiePolicyComponent} />
                                <Route path="/data_collection" component={DataCollectionComponent} />


                            </Switch>
                            </section>
                    
                        </main>
                    </Col>
                </Row>
            </Col>
        </Row>
        <Row className="d-flex theme-dark">
            <Col xs="12">
                <FooterComponent />
            </Col>
        </Row>
      </Container>
    );
  };

