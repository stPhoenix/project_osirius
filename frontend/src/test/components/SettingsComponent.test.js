
import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {SettingsComponent} from '../../components';


describe('SettingsComponent', () => {
    it('should render correctly', () => {
        const componentProps = {
                                handleChange: jest.fn(),
                                username: "",
                                email: "",
                                toggle_modal: jest.fn(),
                                first_name: "",
                                language_set: [],
                                current_language: "",
                                set_active: jest.fn(),
                                delete_language: jest.fn(),
                                langs: [],
                                add_learn_language: jest.fn(),
                                click: jest.fn(),
                                change_password: jest.fn(),
                                password1: "",
                                password2: "",
                                password: "",
                                modal: false,
                                };
        const output = shallow(<SettingsComponent {...componentProps} />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});