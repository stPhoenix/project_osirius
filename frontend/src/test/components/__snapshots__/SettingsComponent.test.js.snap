// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsComponent should render correctly 1`] = `
<Component>
  <form
    className="col-12 col-md-8 col-lg-6 d-flex flex-column h-100 mb-4"
  >
    <div
      className="m-2 form-group"
    >
      <label
        htmlFor="username"
      >
        Change username
      </label>
      <input
        className="custom-form-control"
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
    </div>
    <div
      className="m-2 form-group"
    >
      <label
        htmlFor="email"
      >
        Change email
      </label>
      <input
        className="custom-form-control"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value=""
      />
    </div>
    <button
      className="custom-btn theme-primary align-self-center px-3"
      onClick={[Function]}
    >
      Change password
    </button>
    <div
      className="m-2 form-group"
    >
      <label
        htmlFor="firstname"
      >
        Change your name
      </label>
      <input
        className="custom-form-control"
        id="firstname"
        name="first_name"
        onChange={[Function]}
        placeholder="Your name"
        type="text"
        value=""
      />
    </div>
    <div
      className="m-2 form-group"
    >
      <label
        htmlFor="language_set"
      >
        Learn languages
      </label>
      <div
        className="list-group"
        id="language_set"
      />
    </div>
    <div
      className="mx-2 mt-2 mb-1 form-group"
    >
      <label
        htmlFor="learn_language"
      >
        Add learn language
      </label>
      <select
        className="custom-form-control"
        id="learn_language"
        name="temp_language"
        onChange={[Function]}
      />
    </div>
    <button
      className="custom-btn theme-primary px-3 mb-2 mr-2 align-self-end"
      onClick={[Function]}
    >
      Add
    </button>
    <div
      className="m-2 form-group"
    >
      <label
        className="m-checkbox-container d-flex align-items-center"
      >
        <input
          name="terms_check"
          type="checkbox"
        />
        <span
          className="m-checkbox"
        />
        <h6
          className="ml-4 mb-0 pl-1"
        >
          I agree with 
          <Link
            replace={false}
            to="/privacy_policy"
          >
            Privacy policy
          </Link>
        </h6>
      </label>
    </div>
    <button
      className="btn-danger custom-btn align-self-start px-3 mb-2 ml-2"
    >
      Delete user
    </button>
    <button
      className="custom-btn theme-primary align-self-center px-3"
      onClick={[Function]}
    >
      Save
    </button>
    <Component
      change_password={[Function]}
      handleChange={[Function]}
      modal={false}
      password=""
      password1=""
      password2=""
      toggle_modal={[Function]}
    />
  </form>
</Component>
`;
